src/main/resources/lib/app-metafields/processor/getMetaData.ts

```diff
@@ -57,75 +57,75 @@ function _resolveMetadata(params: MetaDataModel, selfClosingTags=false) {
 
 
 export function getMetaData({
-	appOrSiteConfig,
+	mergedConfig,
 	content=undefined,
 	returnType="json",
 	selfClosingTags=false,
-	siteOrNull,
+	site,
 }: GetMetaDataParams): MetaDataModel|string|undefined {
 	if (!content) {
 		return undefined;
 	}
 
-	const isFrontpage = siteOrNull?._path === content._path;
+	const isFrontpage = site._path === content._path;
 	const pageTitle = getPageTitle({
-		appOrSiteConfig,
+		mergedConfig,
 		content,
 	});
-	const siteVerification = appOrSiteConfig.siteVerification || null;
+	const siteVerification = mergedConfig.siteVerification || null;
 
-	const absoluteUrl = appOrSiteConfig.baseUrl
+	const absoluteUrl = mergedConfig.baseUrl
 	? prependBaseUrl({
-		baseUrl: appOrSiteConfig.baseUrl,
+		baseUrl: mergedConfig.baseUrl,
 		contentPath: content._path,
-		sitePath: siteOrNull?._path || ''
+		sitePath: site._path
 	})
 	: pageUrl({ path: content._path, type: "absolute" });
 
 	const canonicalContent = getContentForCanonicalUrl(content);
 	const canonicalUrl = canonicalContent
-		? appOrSiteConfig.baseUrl
+		? mergedConfig.baseUrl
 			? prependBaseUrl({
-				baseUrl: appOrSiteConfig.baseUrl,
+				baseUrl: mergedConfig.baseUrl,
 				contentPath: canonicalContent._path,
-				sitePath: siteOrNull?._path || ''
+				sitePath: site._path
 			})
 			: pageUrl({ path: canonicalContent._path, type: "absolute" })
 		: null;
 
-	const imageUrl = !appOrSiteConfig.removeOpenGraphImage
+	const imageUrl = !mergedConfig.removeOpenGraphImage
 		? getImageUrl({
-			appOrSiteConfig,
+			mergedConfig,
 			content,
-			defaultImg: appOrSiteConfig.seoImage,
-			defaultImgPrescaled: appOrSiteConfig.seoImageIsPrescaled,
-			siteOrNull,
+			defaultImg: mergedConfig.seoImage,
+			defaultImgPrescaled: mergedConfig.seoImageIsPrescaled,
+			site,
 		})
 		: null;
 
-	const twitterImageUrl = !appOrSiteConfig.removeTwitterImage
+	const twitterImageUrl = !mergedConfig.removeTwitterImage
 		? getImageUrl({
-			appOrSiteConfig,
+			mergedConfig,
 			content,
-			defaultImg: appOrSiteConfig.seoImage,
-			siteOrNull,
+			defaultImg: mergedConfig.seoImage,
+			site,
 		})
 		: null;
```

Comment:
Same here - very similar to the code we have in the widget (`seo.ts`). Refactor?

Reply:
At least those both run in portal "context", but seo can be outside site.
And they again return different things at different points.

I tried to change the processor and seo widget as little as possible, since the initial task was to add Guillotine Extension. In order to refactor these more, I would have to change the seo model and view too.
So It's not a small refactor. I suggest that be done as a separate task. Not as part of extending Guillotine.